### PR TITLE
First implementation of update identity flow

### PR DESCRIPTION
**Pull request by Oluane**
## Description

This PR is a first draft of the update identity flow. The goal was to use a maximum of existing workflows to assemble an update one. 

So the flow starts like a basic add identity one, where we send a validation code, we then redirect to the validation code form and check the validity of the new identity before sending it to Management. 

To compele this flow, just after sending the new identity to Management, we delete the ancient one. To do so, I added an optional key (`identityToUpdateId`) to the `TemporaryIdentity` object that we store in our user table in DynamoDB. This key value is the id of the identity the user wants to update, and we use it to retrieve the old identity value and delete it. 

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [x] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

File: .env_vars.sample [hunk 1]
```diff
@@ -32,6 +32,7 @@ export LIGHTSTEP_ACCESS_TOKEN=
 
 ## Sentry
 export NEXT_PUBLIC_SENTRY_DSN=https://6baf0213093a428d9e9ec7ae0979868c@o287463.ingest.sentry.io/5387583
+export SENTRY_ENVIRONMENT=
```
Comment (Fenntasy) [inline .env_vars.sample:35]:
```suggestion
export SENTRY_ENVIRONMENT= development
```

no?

File: src/pages/api/auth-connect/send-identity-validation-code.ts [hunk 2]
```diff
@@ -55,16 +56,24 @@ const handler: Handler = (request, response): Promise<void> => {
       })
         .then(async ({ eventId }) => {
           span.setDisclosedAttribute("is validation code sent", true);
-
-          const temporaryIdentity = {
+          console.log("bqh lol");
```
Comment (Fenntasy) [inline src/pages/api/auth-connect/send-identity-validation-code.ts:59]: 🚒

File: src/pages/api/auth-connect/send-identity-validation-code.ts [hunk 3]
```diff
@@ -55,16 +56,24 @@ const handler: Handler = (request, response): Promise<void> => {
...
 
+          if (identityToUpdateId) {
+            temporaryIdentity = {
+              ...temporaryIdentity,
+              identityToUpdateId,
+            };
+          }
+
           await insertTemporaryIdentity(userCookie.sub, temporaryIdentity);
+          console.log("rip");
```
Comment (Fenntasy) [inline src/pages/api/auth-connect/send-identity-validation-code.ts:76]: 🧯

**Review by Fenntasy — COMMENTED** [review]
Am I correct when thinking this could move the primary identity if we have 2 identies when updating the primary?

File: src/pages/api/auth-connect/send-identity-validation-code.ts (hunk 2 above)
Comment (Fuzznimp) [reply src/pages/api/auth-connect/send-identity-validation-code.ts:59]: As you can tell by the "q", these were mine..

File: .env_vars.sample (hunk 1 above)
Comment (Fuzznimp) [reply .env_vars.sample:35]: Now that I think about it, the line
```ts
enabled: process.env.NODE_ENV === "production",
```
should prevent any report in dev mode. That's why I did not add the variable in our local `env_vars`.

**Comment by Oluane** [comment]
> Am I correct when thinking this could move the primary identity if we have 2 identies when updating the primary?

I'm not sure to clearly understand what you mean, but I can tell you that if you try to update the primary, the new one will be the new primary in the end.

File: src/pages/api/auth-connect/send-identity-validation-code.ts (hunk 2 above)
Comment (Oluane) [reply src/pages/api/auth-connect/send-identity-validation-code.ts:59]: Oops 😳

File: src/pages/api/auth-connect/send-identity-validation-code.ts (hunk 3 above)
Comment (Oluane) [reply src/pages/api/auth-connect/send-identity-validation-code.ts:76]: 😱 🙇

**Comment by Fenntasy** [comment]
> I'm not sure to clearly understand what you mean, but I can tell you that if you try to update the primary, the new one will be the new primary in the end.

Even if you have:
- identity A, primary
- identity B, created in second
- update A to C 

?

**Comment by Oluane** [comment]
> Even if you have:
> 
>     * identity A, primary
> 
>     * identity B, created in second
> 
>     * update A to C
> 
> 
> ?

Yep, when setting the `TemporaryIdentity` object for the new Identity, I use the `primary` value from the one we want to update. So, in `verify-validation-code.ts`, after `addIdentityToUser` command, it triggers `markIdentityAsPrimary` command and sets the new as primary. (Management handles on its own the switch between primary).

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**